flowpipe/graph.py

```diff
@@ -162,6 +163,51 @@ def _evaluate_threaded(self, submission_delay, raise_after_loops=None):
                 break
             time.sleep(submission_delay)
 
+    def evaluate_multiprocessed(self, submission_delay=0.1):
+        """Similar to the threaded evaluation but with multiprocessing.
+
+        Nodes communicate via a manager and are evaluated in a dedicated
+        function.
+        The original node objects are updated with the results from the
+        corresponding processes to reflect the evaluation.
+        """
+        manager = Manager()
+        nodes_data = manager.dict()
+        processes = {}
+        nodes_to_evaluate = list(self.evaluation_sequence)
+
+        while True:
+            for node in nodes_to_evaluate:
+
+                process = processes.get(node.name)
+                if process and not process.is_alive():
+                    # If the node is done computing, drop it from the list
+                    nodes_to_evaluate.remove(node)
+                    update_node(node, nodes_data[node.identifier])
+                    continue
+
+                upstream_ready = True
```

Comment:
If you move the check into a small function you can get away without the state variable `upstream_ready` and even make it evaluate only when the process is not running yet:
```
def upstream_ready(processes, node):
    for n in node.upstream_nodes:
        p = processes.get(n.name)
        if not p or p.is_alive():
            return False
    return True
```
And then in l. 196:
```
if node.name not in processes and upstream_ready(processes, node):
...
```
Since python evalutes `if`s lazily, this will run `upstream_ready` while a given process is running.

Reply:
Thanks for that recommendation, I added it now